peps/pep-0726.rst

```diff
@@ -209,6 +247,54 @@ attribute access, which is much more likely scenario to get a
 performance penalty.
 
 
+Discussion
+==========
+
+As pointed out by Victor Stinner, the proposed API could be useful already in
+the stdlib, for example to ensure that :py:obj:`sys.modules` type is always a
+list:
+
+.. code:: pycon
+
+   >>> import sys
+   >>> sys.modules = 123
+   >>> import asyncio
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+     File "<frozen importlib._bootstrap>", line 1260, in _find_and_load
+   AttributeError: 'int' object has no attribute 'get'
+
+or to prevent deletion of critical :py:mod:`sys` attributes, which makes the
+code more complicated.  For example, code using :py:obj:`sys.stderr` has to
+check if the attribute exists and if it's not :py:obj:`None`.  Currently, it's
+possible to remove any :py:mod:`sys` attribute, including functions:
+
+.. code:: pycon
+
+   >>> import sys
+   >>> del sys.excepthook
+   >>> 1+  # notice the next line
+   sys.excepthook is missing
+    File "<stdin>", line 1
+      1+
+       ^
+   SyntaxError: invalid syntax
+
+See `related issue
+<https://github.com/python/cpython/issues/106016#issue-1771174774>`__ for
+other details.
+
+Other stdlib modules also come with attributes which can be overriden (as a
+feature) and some input validation here could be helpful.  Examples:
+:py:obj:`threading.excepthook`, :py:obj:`warnings.showwarning`,
+:py:obj:`io.DEFAULT_BUFFER_SIZE` or :py:obj:`os.SEEK_SET`.
+
+There are some constants which may be interesting to implement as constants:
```

Comment:
This "may be interesting" paragraph might be read as a proposed change, these examples are things we'd likely never "protect" from change as these have been replaceable of del-able forever - changing that is likely to break someone's existing code. Let people shoot their own foot. They do serve as good examples of "what could've been done" but the point I'd aim to make is that we could choose to protect newly added things if we see a strong reason to do so. I don't expect it to be something we encourage or do often in the stdlib.

I think your earlier paragraph already effectively makes via practical examples of where it could prove actually useful rather than these hypothetical most likely unimportant ones.

(so maybe just delete this paragraph?)

Reply:
ok